src/ir_York.h

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Daniele Gobbetti
+
+/// @file
+/// @brief Support for the York AC protocol (remote GRYLH2A)
+
+// Supports:
+//   Brand: York,  Model: MHH07P17
+
+#ifndef IR_YORK_H_
+#define IR_YORK_H_
+
+#define __STDC_LIMIT_MACROS
+#include <stdint.h>
+#ifndef UNIT_TEST
+#include <Arduino.h>
+#endif
+#include "IRremoteESP8266.h"
+#include "IRsend.h"
+#ifdef UNIT_TEST
+#include "IRsend_test.h"
+#endif
+
+
+/// Native representation of a York A/C message.
+union YorkProtocol{
+  uint8_t raw[kYorkStateLength];  ///< The state of the IR remote.
+  struct {
+    // byte 0-5
+    uint8_t preamble[6];  // unknown, fixed 0x08, fixed 0x10, fixed 0x07, fixed 0x02, fixed 0x40, fixed 0x08
+    // byte 6
+    uint8_t Key1     :4;  // key pressed on the remote: 1 power, 2 temp up, 3 temp down...
+    uint8_t Key2     :4;  // only set when setting ontime/offtime: Key1 value is 0x6 (enter key) and Key2 is 0x3 for "start" and 0x2 for "stop"
+    //byte 7
+    uint8_t Fan      :4;  // Fan speed: 1 low, 2 mid, 3 max, 8 auto
+    uint8_t Power    :4;  // main unit power: 1 on, 0 off
+    //byte 8
+    uint8_t Mode     :8;  // 1 heat, 2 cool, 3 dry, 4 fan, 8 auto
+    //byte 9
+    uint8_t Temp     :8;  // Degrees Celsius x4: Min temp 18C 0x48 Max temp 32C 0x80
+    //byte 10
+    uint8_t OffTime  :8;  // Power off time: 10s of minutes from now
+    //byte 11
+    uint8_t OnTime   :8;  // Power on time: 10s of minutes from now
+    //byte 12
+    uint8_t          :8;  // unknown, normalyy 0x00, 0x08 when ontime set, 0x88 if both on and offtime set (could mean "auto power")
+    //byte 13
+    uint8_t SwingV   :8;  // 0 off, 1 on
```

Comment:
If those are the only possible values, then it is likely that `SwingV` is a single bi.
so use:
```c++
    uint8_t SwingV   :1;  // 0 off, 1 on
    uint8_t          :7;
```